solr/core/src/test/org/apache/solr/core/ExitableDirectoryReaderTest.java

```diff
@@ -98,7 +102,15 @@ public void testCacheAssumptions() throws Exception {
 
     // This gets 0 docs back. Use 10000 instead of 1 for timeAllowed and it gets 100 back and the for loop below
     // succeeds.
-    String response = JQ(req("q", "*:*", "fq", fq, "indent", "true", "timeAllowed", "1", "sleep", sleep));
+    // TODO: address crosstalk between test methods; failures here can be triggered by cache consultation
```

Comment:
Is this still an issue after you added the special MatchAllDocs cases?

Reply:
Yes, even more so really, since the special-casing for MatchAllDocs currently just throws it to filterCache. Now that I write this, it occurs to me that if special-casing matchAllDocs maybe we can special-case it to _bypass_ the filterCache entirely. I'll give that a try ...